lib/access_token_agent/connector.rb

```diff
@@ -10,20 +12,32 @@ def initialize(host:,
       @client_id = client_id
       @client_secret = client_secret
       @fake_auth = fake_auth
+      set_default_config!
     end
 
-    def authenticate
+    def http_auth_header
+      { Authorization: "Bearer #{token}" }
+    end
+
+    def token
       return if @fake_auth
-      fetch_token unless @known_token && @known_token.valid?
+      fetch_token! unless @known_token && @known_token.valid?
```

Comment:
I wanted to mark a method with a possibly surprising side effect.
I do not buy the "purely functional" argument, since this would also apply to "cached" getters.
I would also not mark the public `token` method, since nobody needs to be afraid of calling that one.

But really look at the code around this comment:
* we only call this method depending on the state of a random instance variable
* after the call, we use that random instance variable again, knowing that this method will have changed the instance variable.

What do you think about this alternative:

* do not assign `known_token` inside `fetch_token`
* assign it here instead, because this is the only place dealing with that instance variable